src/main/java/graphql/normalized/ExecutableNormalizedOperationFactory.java

```diff
@@ -218,14 +215,12 @@ public CollectNFResult collectFromMergedField(FieldCollectorNormalizedQueryParam
                                                   ExecutableNormalizedField executableNormalizedField,
                                                   ImmutableList<FieldAndAstParent> mergedField,
                                                   int level) {
-        GraphQLUnmodifiedType fieldType = unwrapAll(executableNormalizedField.getType(parameters.getGraphQLSchema()));
-        // if not composite we don't have any selectionSet because it is a Scalar or enum
-        if (!(fieldType instanceof GraphQLCompositeType)) {
+        List<GraphQLFieldDefinition> fieldDefs = executableNormalizedField.getFieldDefinitions(parameters.getGraphQLSchema());
```

Comment:
Fixed a bug here where we weren't considering ALL the field definitions. There's a test case for this.

https://github.com/graphql-java/graphql-java/blob/6e2218b7fe66d9b1003e0a863b180b2ebc622786/src/test/groovy/graphql/normalized/ExecutableNormalizedOperationFactoryTest.groovy#L747-L791

Without this fix, the test would've returned

```
-Query.animal: Animal
--[Cat, Dog].parent: Cat, Dog
---[Cat].name: String
```

Instead of

```
-Query.animal: Animal
--[Cat, Dog].parent: Cat, Dog
---[Cat, Dog].name: String 
```